File: src/mk2/JavaPackage/src/main/java/tamaf/ModifyAgentEMABehavior.java
package tamaf;

public class ModifyAgentEMABehavior extends OneShotBehavior {
    public ModifyAgentEMABehavior() {
        super();
    }

    public ModifyAgentEMABehavior(Agent agent) {
        super(agent);
    }

    @Override
    public void action() {
        emaInteraction.modifyAgent();
    }
}

File: src/mk3/esp32package/TAMAF-ESP32/src/tamaf/behaviors/ModifyAgentEMABehavior.cpp
#include "ModifyAgentEMABehavior.h"
#include "tamaf/core/Agent.h"

namespace tamaf {
namespace behaviors {

ModifyAgentEMABehavior::ModifyAgentEMABehavior() 
    : OneShotBehavior(), emaInteraction(this) {}

ModifyAgentEMABehavior::ModifyAgentEMABehavior(tamaf::core::Agent* agent) 
    : OneShotBehavior(agent), emaInteraction(this) {}

void ModifyAgentEMABehavior::Action() {
    emaInteraction.ModifyAgent();
}

} // namespace behaviors
} // namespace tamaf
